src/logger/error.hpp

```diff
@@ -0,0 +1,24 @@
+/* Copyright © 2018 Apple Inc. All rights reserved.
+ *
+ * Use of this source code is governed by a BSD-3-clause license that can
+ * be found in the LICENSE.txt file or at https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/*
+ * Error types to distinguish Turi Create errors from arbitrary C++ exceptions
+ */
+
+#include <ios>
+
+namespace turi {
+    namespace error {
+        class io_error: public std::ios_base::failure {
+        private:
+            std::string m_message;
```

Comment:
I also with we had a consistent way of naming class data members. Honestly, anything is better than having data members be named the same way as local variables, but I think I've seen more instances of `data_member_` than `m_data_member`. This is more food for thought for later, I'm happy this is just not `message`.